### PR TITLE
add ristretto255 docs

### LINKED ISSUE #61: Flesh out all rustdoc in `frost-ristretto255` public API
Half of the public items are well-doc'd, half just have placeholders to allow compilation.

### PR DESCRIPTION
**Pull request by conradoplg**
Based on #108 

Closes #61 

I just copied the docs from frost-core with some simplifications. Let me know if that's OK and if it's better to mention Ristretto255 explicitly, or if that's just implied.

After we decide on this I can write a PR for P-256.

### CODE REVIEW

**Comment by dconnolly** [comment]
> Based on #108 
> 
> Closes #61 
> 
> I just copied the docs from frost-core with some simplifications. Let me know if that's OK and if it's better to mention Ristretto255 explicitly, or if that's just implied.
> 
> After we decide on this I can write a PR for P-256.

Might want to `s/FROST/FROST(ristretto255, SHA-512)/` everywhere that is appropriate and basically do the same for P-256 ✍️

**Comment by conradoplg** [comment]
> Might want to `s/FROST/FROST(ristretto255, SHA-512)/` everywhere that is appropriate and basically do the same for P-256 writing_hand

Done in df865bef15

File: frost-ristretto255/src/lib.rs [hunk 1]
```diff
@@ -198,28 +199,51 @@ pub mod keys {
         frost::keys::keygen_with_dealer(num_signers, threshold, &mut rng)
     }
 
+    /// Secret and public key material generated by a dealer performing
+    /// [`keygen_with_dealer`].
     ///
+    /// To derive a FROST(ristretto255, SHA-512) keypair, the receiver of the [`SharePackage`] *must* call
+    /// .into(), which under the hood also performs validation.
```
Comment (dconnolly) [inline frost-ristretto255/src/lib.rs:206]:
```suggestion
    /// # Safety
    ///
    /// To derive a FROST(ristretto255, SHA-512) keypair, the receiver of the [`SharePackage`] *must* call
    /// .into(), which under the hood also performs validation.
```

Comment (dconnolly) [reply frost-ristretto255/src/lib.rs:206]: I don't know if cryptographic/security safety is 'the same' as memory unsafety in the rust doc sense, but it's still about requirements the API user must adhere to:

File: frost-ristretto255/src/lib.rs [hunk 2]
```diff
@@ -198,28 +199,51 @@ pub mod keys {
...
 }
 
-///
+/// FROST(ristretto255, SHA-512) Round 1 functionality and types.
 pub mod round1 {
     use frost_core::frost::keys::SecretShareValue;
 
     use super::*;
+
+    /// Comprised of hiding and binding nonces.
```
Comment (dconnolly) [inline frost-ristretto255/src/lib.rs:230]:
```suggestion
    /// Comprised of FROST(ristretto255, SHA-512) hiding and binding nonces.
```

File: frost-ristretto255/src/lib.rs [hunk 3]
```diff
@@ -232,20 +256,30 @@ pub mod round1 {
...
+/// each signing party.
 pub type SigningPackage = frost::SigningPackage<R>;
 
-///
+/// FROST(ristretto255, SHA-512) Round 2 functionality and types, for signature share generation.
 pub mod round2 {
     use super::*;
 
-    ///
+    /// A participant's signature share, which the coordinator will aggregate with all other signer's
```
Comment (dconnolly) [inline frost-ristretto255/src/lib.rs:267]:
```suggestion
    /// A FROST(ristretto255, SHA-512) participant's signature share, which the Coordinator will aggregate with all other signer's
```

File: frost-ristretto255/src/lib.rs [hunk 4]
```diff
@@ -255,10 +289,24 @@ pub mod round2 {
     }
 }
 
-///
+/// A Schnorr signature on FROST(ristretto255, SHA-512).
 pub type Signature = frost_core::Signature<R>;
 
+/// Verifies each participant's signature share, and if all are valid,
```
Comment (dconnolly) [inline frost-ristretto255/src/lib.rs:295]:
```suggestion
/// Verifies each FROST(ristretto255, SHA-512) participant's signature share, and if all are valid,
```

File: frost-ristretto255/src/lib.rs [hunk 5]
```diff
@@ -255,10 +289,24 @@ pub mod round2 {
...
+/// aggregates the shares into a signature to publish.
 ///
+/// Resulting signature is compatible with verification of a plain SpendAuth
```
Comment (dconnolly) [inline frost-ristretto255/src/lib.rs:298]:
```suggestion
/// Resulting signature is compatible with verification of a plain Schnorr 
```

File: frost-ristretto255/src/lib.rs (hunk 1 above)
Comment (conradoplg) [reply frost-ristretto255/src/lib.rs:206]: Maybe "Security" would be more appropriate?

**Review by dconnolly — APPROVED** [review]

**Pull request merged**